src/portfft/committed_descriptor_impl.hpp

```diff
@@ -428,8 +423,13 @@ class committed_descriptor_impl {
       }
       return false;
     };
-    detail::factorize_input(fft_size, check_and_select_target_level);
-    return {detail::level::GLOBAL, param_vec};
+    if (!detail::factorize_input(fft_size, check_and_select_target_level)) {
+      param_vec.clear();
+      fft_size = static_cast<IdxGlobal>(std::pow(2, ceil(log(static_cast<double>(fft_size)) / log(2.0))));
```

Comment:
I think using floating-point arithmatic on integer data is a mistake, especially when we have an integer log2 function [here](https://github.com/codeplaysoftware/portFFT/blob/d5fe215de93e38244278909d5f2da8c5853e566f/src/portfft/common/helpers.hpp#L178) that you could adapt.

Reply:
this operation could also be a named function like `round_up_to_pow_2`.